File: VB/WinSolution.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl

Namespace WinSolution.Module
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal session As Session, ByVal currentDBVersion As Version)
			MyBase.New(session, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			Dim p As New Person(Session)
			p.FirstName="Person1"
			Dim pg As New ProjectGroup(Session, "ProjectGroup1")
			pg.Person = p
			Dim pr As New Project(Session, "Project1")
			pr.Person = p
			pg.Projects.Add(pr)
			Dim pa As New ProjectArea(Session, "ProjectArea1")
			pa.Person = p
			pr.ProjectAreas.Add(pa)
			p.Save()
			pa.Save()
			pr.Save()
			pg.Save()
		End Sub
	End Class
End Namespace

File: CS/WinSolution.Module/Updater.cs
using System;

using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;

namespace WinSolution.Module {
    public class Updater : ModuleUpdater {
        public Updater(DevExpress.ExpressApp.IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            Person p1 = ObjectSpace.CreateObject<Person>();
            p1.FirstName="Person1";
            Person p2 = ObjectSpace.CreateObject<Person>();
            p2.FirstName = "Person2";
            ProjectGroup pg = ObjectSpace.CreateObject<ProjectGroup>();
            pg.Name = "ProjectGroup1";
            pg.Person = p1;
            Project pr = ObjectSpace.CreateObject<Project>();
            pr.Name = "Project1";
            pr.Person = p1;
            pg.Projects.Add(pr);
            ProjectArea pa = ObjectSpace.CreateObject<ProjectArea>();
            pa.Name = "ProjectArea1";
            pa.Person = p1;
            pr.ProjectAreas.Add(pa);
            p1.Save();
            p2.Save();
            pa.Save();
            pr.Save();
            pg.Save();
        }
    }
}
